### PR TITLE
Ensure number of realizations output by the apply_emos_coefficients CLI matches the input

### PR DESCRIPTION
**Pull request by Katie-Howard**
Addresses #GitHubissuenum

Remove the use of the realizations_count option when re-creating the realizations within the apply_emos_coefficients cli.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ improver/cli/apply_emos_coefficients.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5006207&bid=16514998)

File: improver/cli/apply_emos_coefficients.py [hunk 1]
```diff
@@ -189,27 +176,13 @@ def process(cube: cli.inputcube,
...
         current_forecast = (
             RebadgePercentilesAsRealizations().process(current_forecast))
 
     # Default number of ensemble realizations is the number in
     # the raw forecast.
-    if not realizations_count:
-        realizations_count = len(
-            current_forecast.coord('realization').points)
-
+    no_of_percentiles = len(current_forecast.coord('realization').points)
```
Comment (cgsandford) [inline improver/cli/apply_emos_coefficients.py]: Please update or remove comment.

Comment (cgsandford) [reply improver/cli/apply_emos_coefficients.py]: Actually that applies to every comment in this file.  Most of them are either misleading or wrong.

Comment (cgsandford) [inline improver/cli/apply_emos_coefficients.py]: This variable isn't used until line 216, and only within that scope.  Please define variable within scope.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ improver/cli/apply_emos_coefficients.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5006207&bid=16514998)

**Pull request merged**